core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSource.scala

```diff
@@ -553,17 +608,25 @@ case class DeltaSource(
    * Get the changes starting from (startVersion, startIndex). The start point should not be
    * included in the result.
    *
-   * If verifyMetadataAction = true, we will break the stream when we detect any read-incompatible
-   * metadata changes.
+   * @param endOffset If defined, do not return changes beyond this offset.
+   *                  If not defined, we must be scanning the log to find the next offset.
+   * @param verifyMetadataAction If true, we will break the stream when we detect any
+   *                             read-incompatible metadata changes.
    */
   protected def getFileChanges(
       fromVersion: Long,
       fromIndex: Long,
       isStartingVersion: Boolean,
-      verifyMetadataAction: Boolean = true): ClosableIterator[IndexedFile] = {
+      endOffset: Option[DeltaSourceOffset] = None,
+      verifyMetadataAction: Boolean = true
+  ): ClosableIterator[IndexedFile] = {
 
     /** Returns matching files that were added on or after startVersion among delta logs. */
     def filterAndIndexDeltaLogs(startVersion: Long): ClosableIterator[IndexedFile] = {
+      // TODO: handle the case when failOnDataLoss = false and we are missing change log files
```

Comment:
Can you crease an issue for this TODO?

Reply:
I have it noted down myself, it's a minor follow up so won't take another issue I think.